pkg/retriever/graphsyncretriever_test.go

```diff
@@ -3,353 +3,452 @@ package retriever
+						events.Started(startTime, retrievalID, startTime, types.RetrievalPhase, types.RetrievalCandidate{MinerPeer: peer.AddrInfo{ID: peer.ID("baz")}}),
+					},
+				},
+				{
+					AfterStart:         time.Millisecond * 20,
+					ReceivedRetrievals: []peer.ID{"foo"},
+					ExpectedEvents: []types.RetrievalEvent{
+						events.Connected(startTime.Add(time.Millisecond*20), retrievalID, startTime, types.RetrievalPhase, types.RetrievalCandidate{MinerPeer: peer.AddrInfo{ID: peer.ID("foo")}}),
+					},
+				},
+				{
+					AfterStart: time.Millisecond * 25,
+					ExpectedEvents: []types.RetrievalEvent{
+						events.Connected(startTime.Add(time.Millisecond*25), retrievalID, startTime, types.RetrievalPhase, types.RetrievalCandidate{MinerPeer: peer.AddrInfo{ID: peer.ID("bar")}}),
+					},
+				},
+				{
+					AfterStart: time.Millisecond * 30,
+					ExpectedEvents: []types.RetrievalEvent{
+						events.Connected(startTime.Add(time.Millisecond*30), retrievalID, startTime, types.RetrievalPhase, types.RetrievalCandidate{MinerPeer: peer.AddrInfo{ID: peer.ID("baz")}}),
+					},
+				},
+				{
+					AfterStart:         time.Millisecond * 120,
+					ReceivedRetrievals: []peer.ID{"bar"},
+					ExpectedEvents: []types.RetrievalEvent{
+						events.Proposed(startTime.Add(time.Millisecond*120), retrievalID, startTime, types.RetrievalCandidate{MinerPeer: peer.AddrInfo{ID: peer.ID("foo")}}),
+						events.Failed(startTime.Add(time.Millisecond*120), retrievalID, startTime, types.RetrievalPhase, types.RetrievalCandidate{MinerPeer: peer.AddrInfo{ID: peer.ID("foo")}}, "retrieval failed: Nope"),
+					},
+				},
+				{
+					AfterStart:         time.Millisecond * 220,
+					ReceivedRetrievals: []peer.ID{"baz"},
+					ExpectedEvents: []types.RetrievalEvent{
+						events.Proposed(startTime.Add(time.Millisecond*220), retrievalID, startTime, types.RetrievalCandidate{MinerPeer: peer.AddrInfo{ID: peer.ID("bar")}}),
+						events.Failed(startTime.Add(time.Millisecond*220), retrievalID, startTime, types.RetrievalPhase, types.RetrievalCandidate{MinerPeer: peer.AddrInfo{ID: peer.ID("bar")}}, "retrieval failed: Nope"),
+					},
+				},
+				{
+					AfterStart: time.Millisecond * 320,
+					ExpectedEvents: []types.RetrievalEvent{
+						events.Proposed(startTime.Add(time.Millisecond*320), retrievalID, startTime, types.RetrievalCandidate{MinerPeer: peer.AddrInfo{ID: peer.ID("baz")}}),
+						events.Failed(startTime.Add(time.Millisecond*320), retrievalID, startTime, types.RetrievalPhase, types.RetrievalCandidate{MinerPeer: peer.AddrInfo{ID: peer.ID("baz")}}, "retrieval failed: Nope"),
+					},
+				},
+			},
 		},
 		// quickest query ("foo") fails retrieval, the other 3 line up in the queue,
-		// it should choose the "best", which is the smallest (they're all free)
+		// it should choose the "best", which is the one that's fast retrieval/verified deal
```

Comment:
I'll be honest, I don't have time now to go over this file and review but I search to see if this test was in here to make up for the removal of all the query sorting testing that's been yeeted; I'll take the rest as good enough if we're getting green CI